pkg/worker/worker_test.go

```diff
@@ -0,0 +1,174 @@
+package worker
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	googleUUID "github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+type MemoryStorage struct {
+	Data map[googleUUID.UUID]Job
+	sync.Mutex
+	MemoryErrors
+}
+
+type MemoryErrors struct {
+	StoreError    error
+	RetrieveError error
+}
+
+func (m *MemoryStorage) Store(job Job) error {
+	m.Data[job.UUID] = job
+	return m.StoreError
+}
+
+func (m *MemoryStorage) Retrieve(uuid googleUUID.UUID) (Job, error) {
+	if job, ok := m.Data[uuid]; ok {
+		return job, m.RetrieveError
+	}
+
+	return Job{}, m.RetrieveError
+}
+
+func TestWorker_Start(t *testing.T) {
+	dataStorage := &MemoryStorage{Data: make(map[googleUUID.UUID]Job)}
+	worker := NewWorker(dataStorage)
+	uuid := worker.Start("echo 'test'", "testUser")
+	var finishedAt time.Time
+
+	for {
+		dataStorage.Lock()
+		finishedAt = dataStorage.Data[uuid].FinishedAt
+		dataStorage.Unlock()
+		if !finishedAt.IsZero() {
+			break
+		}
+	}
```

Comment:
this is a busy wait, which will consume a whole CPU
please add a short sleep in the loop

or, better, add a `job.wait()` helper method